db/db_impl/db_impl.cc

```diff
@@ -823,8 +823,16 @@ Status DBImpl::RegisterRecordSeqnoTimeWorker() {
         min_time_duration / SeqnoToTimeMapping::kMaxSeqnoTimePairsPerCF;
   }
 
-  Status s = periodic_work_scheduler_->RegisterRecordSeqnoTimeWorker(
-      this, seqno_time_cadence);
+  Status s;
+  if (seqno_time_cadence == 0) {
+    periodic_work_scheduler_->UnregisterRecordSeqnoTimeWorker(this);
+  } else {
+    s = periodic_work_scheduler_->RegisterRecordSeqnoTimeWorker(
+        this, seqno_time_cadence);
+  }
+  if (s.ok()) {
+    record_seqno_time_cadence_ = seqno_time_cadence;
```

Comment:
I see the variable set, but didn't see how it used.

Reply:
Yeah, it should be used to compare. Added, the same as above.